Add redirector worker

alexgibson opened the pull request:
- Add redirector worker for https://github.com/mozilla/bedrock/issues/7652.
- Add unit tests.
- Update readme.

alexgibson commented:
@jgmize r?

jgmize commented on workers/redirector.js:
```diff
@@ -0,0 +1,65 @@
+ * The origin that the worker is to be run on.
+ * (e.g. https://www.allizom.org, https://www.mozilla.org).
+ */
+const origin = 'https://www-dev.allizom.org';
```
As discussed, I think it would be better to remove the hard-coded origin so that we can use the same script on any domain.

jgmize commented on test/redirector-test.js:
```diff
@@ -0,0 +1,44 @@
+describe('Redirector Worker', function() {
+
+    it('returns a 200 for requests that have a matching response, but are not within sample rate', async function () {
+        Math.random = sinon.stub().returns(0.6534);
```
Let's use a super low value like 0.0001 here and in the other stubs so that these tests don't break when we lower the sample rate.

jgmize commented on workers/redirector.js:
```diff
@@ -0,0 +1,65 @@
+    {
+        'targetURL': `${origin}/en-US/firefox/new/`,
+        'sandboxURL': `${origin}/en-US/exp/firefox/new/`,
+        'sampleRate': 0.5
```
I believe we wanted to use a lower initial sample rate like 0.05 here.

alexgibson replied:
I talked with Justin and I'm going to set this at ~5%~ 6%

pmac commented:
It seems possible to return the bedrock response directly instead of redirecting. Would that be desirable, or do we for analytics and/or SEO reasons not want to do that?

alexgibson commented:
> It seems possible to return the bedrock response directly instead of redirecting. Would that be desirable, or do we for analytics and/or SEO reasons not want to do that?

I think the answer here is that shadowing could confuse Google since the content would keep changing, but let me double check with Raphael.

alexgibson commented:
Ok, I double checked with Raphael and he says as long as our experimental pages have a canonical that points to the original URLs (which they do), then a 302 is fine. Shadowing is also ok, but only as long as experiments aren't drastically changing the page. Shadowing is a little more risky, since copy changes could essentially get indexed.

bookshelfdave commented on package.json:
```diff
@@ -0,0 +1,24 @@
+    "url": "https://github.com/mozmeao/www-workers/issues"
+  },
+  "devDependencies": {
+    "@dollarshaveclub/cloudworker": "^0.1.1",
```
wow, never thought I'd see dollarshaveclub in a PR :-)

alexgibson commented:
@jgmize I pushed a commit with some review fixes, plus a couple of other small details. Here's a summary:

- Removed `origin` as a variable (switching to relative paths in the config).
- Set sample rate for /en-US/firefox/new/ to 6%.
- Added an additional unit test that checks query params are being preserved on redirect URLs.
- Unit tests now use https://bedrock-stage.gcp.moz.works/ for an origin, so they won't conflict with a real worker running on www-dev etc.
- Added ESLint config check to CI.
- Added `.editorconfig` file.

I've set the updated worker script running on https://www.allizom.org/en-US/firefox/new/ and done some manual testing. It still seems to be working as expecte 👍
